src/uniswap-summaries.md

```diff
@@ -3253,6 +3253,40 @@ module SOLIDITY-UNISWAP-GETRESERVES-SUMMARY
 endmodule
 ```
 
+```k
+module SOLIDITY-MATHSQRT-SUMMARY
+  imports SOLIDITY-CONFIGURATION
+  imports SOLIDITY-UNISWAP-TOKENS
+  imports SOLIDITY-EXPRESSION
+  imports SOLIDITY-STATEMENT
+
+  rule <k> mathSqrt:Id ( v(V:MInt{256}, uint256), .TypedVals ) ~> K => Ss ~> restoreEnv( (y |-> var(size(STORE), uint256)) (z |-> var(size(STORE) +Int 1, uint256)) ) ~> .Statements ~> return z ; ~> .K </k>
+       <summarize> true </summarize>
+       <this-type> TYPE </this-type>
+       <contract-id> TYPE </contract-id>
+       <contract-fn-id> mathSqrt </contract-fn-id>
+       <contract-fn-body> if ( y > 3 ) { z = y ;  uint256 x = y / 2 + 1 ; Ss:Statements } else if ( y != 0 ) { _:Statements }  .Statements </contract-fn-body>
+       <env> E => (y |-> var(size(STORE), uint256))
+                  (z |-> var(size(STORE) +Int 1, uint256))
+                  (x |-> var(size(STORE) +Int 2, uint256))
+       </env>
+       <store> STORE =>
+               STORE ListItem(V) // y
+                     ListItem(V) // z
+                     ListItem((V /uMInt 2p256) +MInt 1p256) // x
+        </store>
+       <current-function> FUNC => mathSqrt </current-function>
```

Comment:
This is necessary because we're not inside the `mathSqrt` yet, right?

Reply:
Yes. We are doing a call, but stopping before it returns. so are including all changes made by the call rule to the `<env>`, `<k>`, `<call-stack>` and we need to change the <current-function> as well.

Reply:
Got it! Thanks!